Toolkit/ArcGISToolkit/BookmarksViewController.swift

```diff
@@ -0,0 +1,171 @@
+        addChild(tableViewController)
+        view.addSubview(tableViewController.view)
+        tableViewController.view.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            tableViewController.view.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            tableViewController.view.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            tableViewController.view.topAnchor.constraint(equalTo: view.topAnchor),
+            tableViewController.view.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+        ])
+        tableViewController.didMove(toParent: self)
+        
+        tableViewController.delegate = self
+    }
+
+    private func geoViewDidChange(_ previousGeoView: AGSGeoView?) {
+        if let mapView = geoView as? AGSMapView {
+            mapView.map?.load { [weak self] (error) in
+                if let error = error {
+                    print("Error loading map: \(error)")
+                }
+                self?.bookmarks = (self?.geoView as? AGSMapView)?.map?.bookmarks as? [AGSBookmark] ?? []
+            }
+            
+            // Add an observer to handle changes to the map.bookmarks array.
+            bookmarksObservation = mapView.map?.observe(\.bookmarks) { [weak self] (map, _) in
+                self?.bookmarks = map.bookmarks as? [AGSBookmark] ?? []
+            }
+            
+            // Add an observer to handle changes to the mapView.map.
+            mapOrSceneObservation = mapView.observe(\.map) { [weak self] (_, _) in
+                self?.bookmarks = mapView.map?.bookmarks as? [AGSBookmark] ?? []
+            }
+        } else if let sceneView = geoView as? AGSSceneView {
+            sceneView.scene?.load { [weak self] (error) in
+                if let error = error {
+                    print("Error loading scene: \(error)")
+                }
+                self?.bookmarks = (self?.geoView as? AGSSceneView)?.scene?.bookmarks as? [AGSBookmark] ?? []
+            }
+            
+            // Add an observer to handle changesto the scene.bookmarks array.
+            bookmarksObservation = sceneView.scene?.observe(\.bookmarks) { [weak self] (scene, _) in
+                self?.bookmarks = scene.bookmarks as? [AGSBookmark] ?? []
+            }
+            
+            // Add an observer to handle changes to the sceneView.scene.
+            mapOrSceneObservation = sceneView.observe(\.scene) { [weak self] (_, _) in
+                self?.bookmarks = sceneView.scene?.bookmarks as? [AGSBookmark] ?? []
+            }
+        }
+    }
+}
+
+// MARK: BookmarksViewControllerDelegate
+extension BookmarksViewController: BookmarksViewControllerDelegate {
+    public func bookmarkSelectionDidChange(_ bookmark: AGSBookmark) {
+        if let delegate = delegate {
+            delegate.bookmarkSelectionDidChange(bookmark)
+        } else if let geoView = geoView, let viewpoint = bookmark.viewpoint {
```

Comment:
I've removed the default behavior of both setting the viewpoint and dismissing the VC because of the issues you mentioned.  The code is much simpler now.